fgpyo/fasta/sequence_dictionary.py

```diff
@@ -0,0 +1,502 @@
+    def __len__(self) -> int:
+        return self.end - self.start + 1
+
+    @staticmethod
+    def parse(value: str) -> "AlternateLocus":
+        """Parse the genomic interval of format: `<contig>:<start>-<end>`"""
+        name, rest = value.split(":", maxsplit=1)
+        start, end = rest.split("-", maxsplit=1)
+        return AlternateLocus(name=name, start=int(start), end=int(end))
+
+
+SEQUENCE_NAME_PATTERN: Pattern = re.compile(
+    "^[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*$"
+)
+"""Regular expression for valid reference sequence names according to the SAM spec"""
+
+
+@dataclass(frozen=True, init=True)
+class SequenceMetadata(MutableMapping[Union[Keys, str], str]):
+    """Stores information about a single Sequence (ex. chromosome, contig).
+
+    Implements the mutable mapping interface, which provide access to the attributes of this
+    sequence, including name, length, but not index.  When using the mapping interface, for example
+    getting, setting, deleting, as well as iterating over keys, values, and items, the _values_ will
+    always be strings (`str` type).  For example, the length will be an `str` when accessing via
+    `get`; access the length directly or use `len` to return an `int`.  Similarly, use the
+    `alias` property to return a `List[str]` of aliases, use the `alternate` property to return
+    an `AlternativeLocus`-typed instance, and `topology` property to return a `Toplogy`-typed
+    instance.
+
+    All attributes except name and length may be set.  Use `dataclasses.replace` to create a new
+    copy in such cases.
+
+    Important: The `len` method returns the length of the sequence, not the length of the
+    attributes.  Use `len(meta.attributes)` for the latter.
+
+    Attributes:
+      name: the primary name of the sequence
+      length: the length of the sequence, or zero if unknown
+      index: the index in the sequence dictionary
+      attributes: attributes of this sequence
+    """
+
+    name: str
+    length: int
+    index: int
+    attributes: Dict[Union[Keys, str], str] = field(default_factory=dict)
+
+    def __post_init__(self) -> None:
+        """Any post initialization validation should go here"""
+        if self.length < 0:
+            raise ValueError(f"Length must be >= 0 for '{self.name}'")
+        if re.search(SEQUENCE_NAME_PATTERN, self.name) is None:
+            raise ValueError(f"Illegal name: '{self.name}'")
+        if Keys.SEQUENCE_NAME in self.attributes:
+            raise ValueError(f"'{Keys.SEQUENCE_NAME}' should not given in the list of attributes")
+        if Keys.SEQUENCE_LENGTH in self.attributes:
+            raise ValueError(f"'{Keys.SEQUENCE_LENGTH}' should not given in the list of attributes")
+
+    @property
+    def aliases(self) -> List[str]:
+        """The aliases (not including the primary) name"""
+        if Keys.ALIASES not in self.attributes:
+            return []
+        return self.attributes[Keys.ALIASES].split(",")
```

Comment:
```suggestion
        aliases = self.attributes.get(Keys.ALIASES)
        return [] if aliases is None else aliases.split(",")
```

To match the form used for `topology`